### PR TITLE
typedbundleprovider getallresources override

### PR DESCRIPTION
**Pull request by Capt-Mac**
HapiFhirDal in Cr-common had a bug where pagination of searched resources resulted in null value if default queryCount was exceeded. Overriding from TypedBundleProvider.getallresources fixed the issue

### CODE REVIEW

**Comment by JPercival** [comment]
Do we have a test that can reproduce the buggy behavior to validate that it's fixed?

In theory, loading up the server with 20+ dummy Patients and then trying to search for them using the HapiFhirDal would repro it.

**Comment by Capt-Mac** [comment]
> 

added test to verify search finds all resources past set queryCount value, old code causes null pointer exception for this test, current code identifies all resources posted from test data

File: hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/r4/CqlMeasureEvaluationR4ImmunizationTest.java [hunk 1]
```diff
@@ -73,4 +90,23 @@ public void test_Immunization_Ontario_Schedule() throws IOException {
 		assertMeasureScore(reportIndividualImmunized, 1.0); // the patient is fully immunized on on 2022-09-16
 		assertMeasureScore(reportIndividualNotImmunized, 0.0); // the patient is not fully immunized on 2022-09-16
 	}
+	@Test
+	void canSearchMoreThan50Patients(){
```
Comment (JPercival) [inline hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/r4/CqlMeasureEvaluationR4ImmunizationTest.java:94]: Could we instead add this to a Test class for the HapiFhirDal? This test isn't actually related to the immunization use case.

**Review by JPercival — CHANGES_REQUESTED** [review]

File: hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/r4/CqlMeasureEvaluationR4ImmunizationTest.java (hunk 1 above)
Comment (Capt-Mac) [reply hapi-fhir-storage-cr/src/test/java/ca/uhn/fhir/cr/r4/CqlMeasureEvaluationR4ImmunizationTest.java]: moved to it's own class

**Review by JPercival — APPROVED** [review]

**Comment by tadgh** [comment]
Since DQM is now publicly available, this change requires a changelog. You can place your changelog in the `rel_6_6_0` folder in the docs project. Just copy an existing one, and modify it.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_6_0/4551-fix-hapifhirdal-search-pagination.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+type: change
```
Comment (tadgh) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_6_0/4551-fix-hapifhirdal-search-pagination.yaml:2]: This type should be `fix` as opposed to change, assuming my read of this as a bug is correct.

**Review by tadgh — APPROVED** [review]
Approved pending one final change

**Pull request merged**